MetaMask/metamask-extension
# Fix incorrect warnings when adding a chain from a dapp

## **Description**

When a user adds a chain from a dapp, they will see warnings that the currency symbol is incorrect, even though it is correct.

Part of the problem was that the `pendingConfirmation` variable inside of `fetchSafeChainsList` was undefined when it should not be. I guess it closed over the original value or something like that? I didn't verify. This PR corrects it by explictly passing the updated `pendingConfirmations` variable to `fetchSafeChainList`

The other problem was that `useAlertState` was calling `getTemplateAlerts` twice, without resetting the alert state. On the first call, alerts were being set in state because the request to chainId.network had not yet resolved. This PR fixes it by ensuring that that `useAlertState` does not call `getTemplateAlerts` before the fetch to chainId.network has resolved (when this confrimation type is an add ethereum chain approval)

## **Related issues**

An example of the problem can be seen by going to https://chainlist.org/?search=cro and adding Cronos Mainnet to metamask, only to see warnings that the currency symbol is incorrect even though it is correct.

![Screenshot from 2023-12-15 18-26-51](https://github.com/MetaMask/metamask-extension/assets/7499938/80e299d4-1545-4a38-981f-8b1eafc0b3dc)


## **Manual testing steps**

1. Go to https://chainlist.org/?search=cro
2. Add Cronos mainnet to metamask
3. The confirmation window should open with an add ethereum chain confirmation. There should be no warnings
4. Reject the confirmation
5. Open the dev console and paste the below code, and press enter. The confirmation window should open with an add ethereum chain confirmation. There should be warnings about the currency symbol
6. Reject the confirmation
7. Modify the below code so that `"symbol": "cra",` becomes `"symbol": "cro",`, paste it in the dev console and press enter. There should be no warnings in the new confirmation window

```
await window.ethereum.request({
  "method": "wallet_addEthereumChain",
  "params": [
    {
      "blockExplorerUrls": [
        "https://blockscout.com/poa/xdai/"
      ],
      "nativeCurrency": {
        "name": "CRO",
        "symbol": "cra",
        "decimals": 18
      },
      "rpcUrls": [
        "https://evm.cronos.org",
        null
      ],
      "chainId": "0x19",
      "chainName": "Cronos Mainnet"
    }
  ]
});
```

### **Before**

https://github.com/MetaMask/metamask-extension/assets/7499938/d6f738be-8e4a-4618-8703-c2d94df9c8a7

### **After**


https://github.com/MetaMask/metamask-extension/assets/7499938/c3ab5cf6-ac73-4ad3-b7b3-fd49910c0311





## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
